src/webxr/VRButton.js

```diff
@@ -130,6 +132,18 @@ class VRButton {
       return message
     }
   }
+
+  static xrSessionIsGranted = false
+
+  static registerSessionGrantedListener() {
+    if ('xr' in navigator) {
+      navigator.xr.addEventListener('sessiongranted', () => {
+        VRButton.xrSessionIsGranted = true
+      })
+    }
+  }
 }
 
+VRButton.registerSessionGrantedListener()
```

Comment:
This is necessary (in combination with browser flags) to enable the immersive navigation proposal. I'm assuming this breaks tree-shaking so I'd want to remove it in that case.

Reply:
It does. This will need to be done in user-land.